src/rws_interface.cpp

```diff
@@ -117,6 +117,111 @@ std::string RWSInterface::getIOSignal(const std::string& iosignal)
   return result;
 }
 
+bool RWSInterface::getMechanicalUnitStaticInfo(const std::string& mechunit, MechanicalUnitStaticInfo* p_static_info)
+{
+  bool result = false;
+
+  if (p_static_info)
+  {
+    RWSClient::RWSResult rws_result = rws_client_.getMechanicalUnitStaticInfo(mechunit);
+
+    if(rws_result.success)
+    {
+      p_static_info->task_name = xmlFindTextContent(rws_result.p_xml_document, XMLAttribute("class", "task-name"));
+      p_static_info->is_integrated_unit = xmlFindTextContent(rws_result.p_xml_document,
+                                                             XMLAttribute("class", "is-integrated-unit"));
+      p_static_info->has_integrated_unit = xmlFindTextContent(rws_result.p_xml_document,
+                                                              XMLAttribute("class", "has-integrated-unit"));
+
+      std::string type = xmlFindTextContent(rws_result.p_xml_document, XMLAttribute("class", "type"));
+      if(type == "None")
+      {
+        p_static_info->type = NONE;
+      }
+      else if(type == "TCPRobot")
+      {
+        p_static_info->type = TCP_ROBOT;
+      }
+      else if(type == "Robot")
+      {
+        p_static_info->type = ROBOT;
+      }
+      else if(type == "Single")
+      {
+        p_static_info->type = SINGLE;
+      }
+      else
+      {
+        p_static_info->type = UNDEFINED;
+      }
+
+      std::stringstream axes(xmlFindTextContent(rws_result.p_xml_document, XMLAttribute("class", "axes")));
+      axes >> p_static_info->axes;
```

Comment:
Not sure what sort of error we get if this conversion fails?

Reply:
Hm... I guess its's possible to use the `stringstream`'s `fail()` function to do some checks.

I am not sure if it will catch everything that could go wrong, but it should at least pass the cases that are fine.

I can add it to a verification step.

Reply:
Included use of `fail()` function in a basic verification in 6570ff0

Reply:
Nice. :+1: